### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by motiz88**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | no
| Deprecations?     | no
| Spec compliancy?  | no
| Tests added/pass? | yes
| License           | MIT

In much the same vein as babel/babylon#175 - this is the first batch of new tests written explicitly to hit previously uncovered lines/branches. I'll be adding more, but they are all independent of one another, so if anyone wants to take time to review & merge some snapshot of this branch - just let me know and I'll move to another one.

### CODE REVIEW

File: packages/babel-traverse/src/path/inference/inferers.js [hunk 1]
```diff
@@ -137,7 +137,13 @@ function Func() {
   return t.genericTypeAnnotation(t.identifier("Function"));
 }
 
-export { Func as Function, Func as Class };
+export {
+  Func as FunctionExpression,
```
Comment (hzoo) [inline packages/babel-traverse/src/path/inference/inferers.js]: Would this be breaking if we remove `Func as Function, Func as Class`? (although maybe it's not used at all)

Comment (motiz88) [inline packages/babel-traverse/src/path/inference/inferers.js]: I interpreted it to be an implementation detail, where `getTypeAnnotation()`, `baseTypeStrictlyMatches()` etc are the actual API.

Comment (motiz88) [inline packages/babel-traverse/src/path/inference/inferers.js]: And IIRC this was actually _broken_ the old way - i.e. whoever wrote this was clearly expecting some code to expand `Function` and `Class` a la visitor method names, and it wasn't happening. (But not tested so it only came up when I tackled coverage)

Comment (hzoo) [inline packages/babel-traverse/src/path/inference/inferers.js]: 👍  yeah we have a lot that might not be used anywhere

**Comment by danez** [comment]
Awesome.

**Pull request merged**